src/graph.py

```diff
@@ -1,242 +1,192 @@
-
-
-class GNode(object):
-    """Create a GNode Constructor."""
-
-    def __init__(self, value):
-        """Create an instance of a GNode with the attribues of a value and a list of edges."""
-        self.value = value
-        self.edges = set()
-
-
-class Edge(object):
-    """Create an Edge Constructor."""
-
-    def __init__(self, begin, end):
-        """Create an instance of an edge."""
-        self.begin = begin
-        self.end = end
-
-    def __eq__(self, other):
-        """Allow comparison of two equal edges."""
-        return self.begin == other.begin and self.end == other.end
-
-    def __ne__(self, other):
-        """Allow comparison of two unequal edges."""
-        return self.begin != other.begin or self.end != other.end
-
-
 class Graph(object):
     """Create a Graph Constructor."""
 
     def __init__(self):
         """Create an instance of a Graph that has a list of nodes."""
+        self.gnodes = {}
         self.set_of_nodes = set()
 
-    def add_node(self, new_node):
+    def add_node(self, value):
         """Add a new node to the graph."""
-        self.is_node(new_node)
-        self.set_of_nodes.add(new_node)
+        if value not in self.set_of_nodes:
+            try:
+                self.gnodes[value] = {}
+                self.set_of_nodes.add(value)
+            except TypeError:
+                raise TypeError('Nodes value need to be a hashable type.')
```

Comment:
Great that you are handling this error. Make sure your are testing for it, however.